### PR TITLE
Add badges for github actions

### PR DESCRIPTION
**Pull request by cody-littley**
# Pull Request Template

## Fixes Issue

[Add icons to eigenda-proxy landing page for github action status.](https://linear.app/eigenlabs/issue/EGDA-715/add-github-action-icons-for-eigenda-proxy)

## Changes proposed

- Broke up "actions" workflow into smaller workflows. You can only have a badge at the workflow granularity (not at the job granularity).
- Added actions icons to the readme file.

### Screenshots (Optional)

To see a view of the README file as it is in this branch, visit https://github.com/cody-littley/eigenda-proxy/tree/eigenda-proxy-badges?tab=readme-ov-file

Screenshot 2024-08-07 at 11 55 29 AM

## Note to reviewers

Icons will not turn green until this merges and the tests run against master.

### CODE REVIEW

**Review by bxue-l2 — APPROVED** [review]

**Review by jianoaix — COMMENTED** [review]
LGTM after the failing test is resolved

**Review by epociask — APPROVED** [review]
LGTM - like how the use of badgs forces proper abstract segmentation of workflow jobs across files. Btw the test is failing due to a regression where forked PRs can't access the github env secrets predefined upstream (i.e, `SIGNER_PRIVATE_KEY environment variable not set`). This will pass once merge.

**Pull request merged**
